modules/claims_api/lib/claims_api/error/soap_error_handler.rb

```diff
@@ -5,79 +5,74 @@
 module ClaimsApi
   class SoapErrorHandler
     # list of fault codes: https://hub.verj.io/ebase/doc/SOAP_Faults.htm
-    # list faraday errors: https://lostisland.github.io/faraday/middleware/raise-error
 
     def handle_errors(response)
       @hash = Hash.from_xml(response.body)
 
       return if @hash&.dig('Envelope', 'Body', 'Fault').blank?
 
-      # catch message and log it with claim id
-      fault_description = get_fault_description
-
-      get_exception(fault_description)
-    end
-
-    def get_fault
-      fault_code = @hash&.dig('Envelope', 'Body', 'Fault', 'faultcode')
-      fault_code&.split(':')&.dig(1)
+      get_fault_info
     end
 
-    def get_fault_description
+    def get_fault_info
+      @fault_code = @hash&.dig('Envelope', 'Body', 'Fault', 'faultcode')&.split(':')&.dig(1)
       @fault_string = @hash&.dig('Envelope', 'Body', 'Fault', 'faultstring')
       @fault_message = @hash&.dig('Envelope', 'Body', 'Fault', 'detail', 'MessageException')
       return {} if @fault_string.include?('IntentToFileWebService') && @fault_string.include?('not found')
 
-      @reason = ''
-      unless person_found? == 'not found' || claim_found? == {} || processable? == 'unprocessable'
-        ClaimsApi::Logger.log("Returning 500 via local_bgs & soap_error_handler, fault_string:#{@fault_string},
-          with message: #{@fault_message}.")
-
-      end
-      @reason
+      get_exception
     end
 
-    def get_exception(fault_description)
-      case fault_description
-      when 'not_found'
+    private
+
+    def get_exception
+      if claim_not_found?
```

Comment:
@stiehlrod  I noticed some unexpected behavior here I wanted to ask about.  When I was testing this I sent in 'No BnftClaim found' in the @fault_string.  When I did that I noticed that the log out from the `clam_not_found?` method still occurred which seemed like it probably was not intended to do that?  I was wondering if the intention was to `return` if has_errors is false on these methods?  when I stepped through and got to line 49 has_errors _was_ false, but it still logged out the next line.  Then the log out occurred again in the `bnft_claim_not_found?`method which was what I expected.

What I saw logged out exactly n case it helps 
```
2023-10-09 14:22:11.197677 I [70782:puma srv tp 002] Rails -- ClaimsApi :: soap_error_handler :: {"detail":"Returning 404 via local_bgs & soap_error_handler, fault_string: No BnftClaim found, with message: , and fault_code: ."} :: Location: /Users/rockwellrice/Documents/oddball/vets-api/modules/claims_api/lib/claims_api/error/soap_error_handler.rb:50
2023-10-09 14:22:11.198729 I [70782:puma srv tp 002] Rails -- ClaimsApi :: soap_error_handler :: {"detail":"Returning 404 via local_bgs & soap_error_handler, fault_string: No BnftClaim found, with message: , and fault_code: ."} :: Location: /Users/rockwellrice/Documents/oddball/vets-api/modules/claims_api/lib/claims_api/error/soap_error_handler.rb:60
```